src/lib/journey/journey.interfaces.ts

```diff
@@ -31,6 +31,7 @@ export interface CallbackMetadata {
   isSelfSubmitting: boolean;
   isUserInputRequired: boolean;
   idx: number;
+  social?: Maybe<SocialMetadata>;
```

Comment:
I figured that when defining metadata around the social aspect of a callback, it would be best for the callback metadata to use a Maybe so that a `SocialMetaData` being present would imply that the callback is a social one (handy to know in other parts of the app), and that we don't necessarily over-burden the metadata with another setting (i.e. every callback explicitly stating that it is / isn't a social callback, and having a mostly superfluous property in indicating local authentication). However I'm working on my TypeScript, so @cerebrl @ryanbas21 I would appreciate your view on how this could best be structured! 😄

Reply:
Is there a reason why we don't just `extend` this interface and create a `SocialMetadata` inferface?

something like 

```
interface SocialMetaDataCallback extends CallbackMetadata {
 social: SocialMetaData
 }
```

Reply:
The suggestion I made earlier should replace this high specificity property with a more generic one: `canForceUserInputOptionality`.

Reply:
Nice, I'm reviewing your suggestions now - I think I had a similar idea in an earlier commit before I went down the social metadata path - the idea of "forcibly self submitting" i.e. ignoring all other user input and self submitting. I'll work through it. Thanks.